db/entity/src/conversions/logs.rs

```diff
@@ -0,0 +1,82 @@
+use sea_orm::Set;
+
+use hopr_crypto_types::types::Hash;
+use hopr_primitive_types::prelude::*;
+
+use crate::errors::DbEntityError;
+use crate::{log, log_status};
+
+impl From<SerializableLog> for log::ActiveModel {
+    fn from(value: SerializableLog) -> Self {
+        log::ActiveModel {
+            address: Set(value.address),
+            topics: Set(value.topics.join(",")),
+            data: Set(value.data),
+            block_number: Set(value.block_number.to_be_bytes().to_vec()),
+            transaction_hash: Set(Hash::from_hex(value.tx_hash.as_str())
+                .expect("invalid tx_hash")
+                .as_ref()
+                .to_vec()),
+            transaction_index: Set(value.tx_index.to_be_bytes().to_vec()),
+            block_hash: Set(Hash::from_hex(value.block_hash.as_str())
+                .expect("invalid block_hash")
+                .as_ref()
+                .to_vec()),
+            log_index: Set(value.log_index.to_be_bytes().to_vec()),
+            removed: Set(value.removed),
+            ..Default::default()
+        }
+    }
+}
+
+impl TryFrom<log::Model> for SerializableLog {
+    type Error = DbEntityError;
+
+    fn try_from(value: log::Model) -> Result<Self, Self::Error> {
+        let tx_hash: Result<[u8; 32], _> = value.transaction_hash.try_into();
+        let block_hash: Result<[u8; 32], _> = value.block_hash.try_into();
+
+        if let Err(_) = tx_hash {
+            return Err(DbEntityError::ConversionError(format!("Invalid tx_hash")));
+        }
```

Comment:
`let tx_hash: [u8; 32] = value.transaction_hash.try_into().map_err(|_| DbEntityError::ConversionError(format!("Invalid tx_hash")))?;`

And you get rid of those `if`s.